### PR TITLE
Adding key field in product

### PR DESCRIPTION
**Pull request by Akhilkrishna11**
This PR is created to Add a new field `Key` while creating Product

### CODE REVIEW

File: src/Kmd.Logic.Gateway.Automation/PublishFile/Product.cs [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 
 namespace Kmd.Logic.Gateway.Automation.PublishFile
 {
     internal class Product
     {
+        private const string Value = "^[a-zA-Z0-9-]*$";
+
         public string Name { get; set; }
 
+        [RegularExpression(Value, ErrorMessage = "Key accepts only Alphanumeric characters and hypen")]
```
Comment (rosieks) [inline src/Kmd.Logic.Gateway.Automation/PublishFile/Product.cs:12]: Does anyone use it?

**Comment by jaceks2106** [comment]
Doesn't this 'key' functionality require any tests?

**Comment by Akhilkrishna11** [comment]
> Doesn't this 'key' functionality require any tests?

We are not having tests in Git Client

File: src/Kmd.Logic.Gateway.Automation/PublishFile/Product.cs (hunk 1 above)
Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/PublishFile/Product.cs]: This validation is removed from here and added in Lab

**Review by rosieks — APPROVED** [review]

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+                }
+            }
+
+            // Deserialize Response
+            if ((int)statusCode == 400)
+            {
+                responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
+                try
+                {
+                    result.Body = SafeJsonConvert.DeserializeObject<IDictionary<string, IList<string>>>(responseContent, this.DeserializationSettings);
```
Comment (nakulnseerpu) [inline src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:645]: Are you sure this will deserialiaze?

Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs]: Removed IList and using string. Now able to deserialize properly

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs [hunk 3]
```diff
@@ -642,7 +642,8 @@ public async Task<HttpOperationResponse<object>> CustomUpdateProductWithHttpMess
                 responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
                 try
                 {
-                    result.Body = SafeJsonConvert.DeserializeObject<IDictionary<string, IList<string>>>(responseContent, this.DeserializationSettings);
+                    result.Body = SafeJsonConvert.DeserializeObject<string>(responseContent, this.DeserializationSettings);
+                    Console.WriteLine(result.Body);
```
Comment (rosieks) [inline src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:646]: Should we print this to output?

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs [hunk 4]
```diff
@@ -663,26 +664,7 @@ public async Task<HttpOperationResponse<object>> CustomUpdateProductWithHttpMess
...
-            }
-
-            // Deserialize Response
-            if ((int)statusCode == 422)
-            {
-                responseContent = await httpResponse.Content.ReadAsStringAsync().ConfigureAwait(false);
-                try
-                {
-                    result.Body = SafeJsonConvert.DeserializeObject<IDictionary<string, IList<string>>>(responseContent, this.DeserializationSettings);
+                    Console.WriteLine(result.Body);
```
Comment (rosieks) [inline src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:667]: Same here - should we print that to console?

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs (hunk 3 above)
Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:646]: We are displaying the appropriate error message. If we are not printing this, then the error message is not displayed in console

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs (hunk 4 above)
Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:667]: We are displaying the appropriate error message. If we are not printing this, then the error message is not displayed in console

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs (hunk 3 above)
Comment (nakulnseerpu) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs:646]: If we print this , we will not be able to figure out for which entity (product) we get error.I think we should make use of "GatewayAutomationResult" to write errors

Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs]: Removed Console.writeline

File: src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs (hunk 4 above)
Comment (Akhilkrishna11) [reply src/Kmd.Logic.Gateway.Automation/Client/Extensions/GatewayClient.ProductExtensions.cs]: Removed Console.writeline

**Review by rosieks — APPROVED** [review]

**Pull request merged**